horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java

```diff
@@ -1004,4 +1004,137 @@ public DatasetList listDatasetsBySchema(String uri, Integer limit, Integer page,
       DatasetList list = new DatasetList();
       return list;
    }
+
+   private void transform(Run run, String role, String token) {
+      if (run == null || role == null) {
+         log.errorf("Transformation parameters error: run %s, role %s", run, role);;
+      }
+      Collection<RunService.RunSchema> runSchemas = getSchemas(run, role, token);
+      DataSet ds = null;
+
+      try (@SuppressWarnings("unused") CloseMe h1 = roleManager.withRoles(em, role);
+            @SuppressWarnings("unused") CloseMe h2 = roleManager.withToken(em, token)) {
+
+         Test test = Test.findById(run.testid);
+         List<JsonNode> nodesToProcess = new ArrayList<JsonNode>();
+         boolean matched = false;
+   
+         if (!runSchemas.isEmpty()){
+            for ( RunSchema s : runSchemas ){
+               findMatches(s, run.data, nodesToProcess, true);
```

Comment:
If there are 2 schemas it seems it keeps adding to `nodesToProcess`, so the second schema would be applied against nodes that don't match it.

Reply:
thanks for catching that.